File: 09.PalindromeNumber/main.go
package main

import (
    "fmt"
    "strconv"
)

func isPalindrome(x int) bool {
    s := strconv.Itoa(x)
    reversed := ""
    
    for i := len(s) - 1; i >= 0; i-- {
        reversed += string(s[i])
    }
    
    return s == reversed
}

func main() {
    fmt.Println(isPalindrome(121))
    fmt.Println(isPalindrome(-121))
    fmt.Println(isPalindrome(10))
}

File: 09.PalindromeNumber/main.c
#include <stdio.h>
#include <stdbool.h>

bool isPalindrome(int x) {
    if (x < 0 || (x % 10 == 0 && x != 0)) return false;
    int reversed = 0;
    while (x > reversed) {
        reversed = reversed * 10 + x % 10;
        x /= 10;
    }
    return x == reversed || x == reversed / 10;
}

int main() {
    printf("%d\n", isPalindrome(121));
    printf("%d\n", isPalindrome(-121));
    printf("%d\n", isPalindrome(10));
    printf("%d\n", isPalindrome(123454321));
    return 0;
}
